spec/classes/collectd_plugin_write_prometheus_spec.rb

```diff
@@ -12,14 +12,51 @@
       options = os_specific_options(facts)
       context ':ensure => present and :port => 9103' do
         let :params do
-          { port: 9103 }
+          {
+            port: 9103,
+            host: "abc"
+          }
         end
 
         it "Will create #{options[:plugin_conf_dir]}/10-write_prometheus.conf" do
           is_expected.to contain_file('write_prometheus.load').with(
             ensure: 'present',
             path: "#{options[:plugin_conf_dir]}/10-write_prometheus.conf",
             content: %r{Port "9103"}
+          ).without(
+            content: %r{Host "abc"}
+          )
+        end
+
+        context "Will include hostname if version is >= 5.9" do
+          let :facts do
+            facts.merge(collectd_version: '5.9')
+          end
+
+          is_expected.to contain_file('write_prometheus.load').with(
+            ensure: 'present',
+            path: "#{options[:plugin_conf_dir]}/10-write_prometheus.conf",
+            content: %r{Port "9103"}
+          ).with(
+            content: %r{Host "abc"}
+          )
```

Comment:
This should be wrapped in a `it` block:

```ruby
it 'description' do
  is_expected.to ...
end
```

Reply:
Yeah, I'm still trying to set up my environment to run the tests properly. Work in progress. I should be back :)

Reply:
Let me know how it's looking now